app/src/main/java/com/nextcloud/talk/ui/dialog/SetStatusDialogFragment.kt

```diff
@@ -0,0 +1,525 @@
+import com.nextcloud.talk.models.json.status.predefined.PredefinedStatus
+import com.nextcloud.talk.models.json.status.Status
+import com.nextcloud.talk.models.json.status.StatusType
+import com.nextcloud.talk.models.json.status.predefined.PredefinedStatusOCS
+import com.nextcloud.talk.models.json.status.predefined.PredefinedStatusOverall
+import com.nextcloud.talk.utils.ApiUtils
+import com.nextcloud.talk.utils.DisplayUtils
+import com.vanniktech.emoji.EmojiPopup
+import io.reactivex.Observer
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import okhttp3.ResponseBody
+import java.util.Calendar
+import java.util.HashSet
+import java.util.Locale
+import javax.inject.Inject
+
+private const val ARG_CURRENT_USER_PARAM = "currentUser"
+private const val ARG_CURRENT_STATUS_PARAM = "currentStatus"
+
+private const val POS_DONT_CLEAR = 0
+private const val POS_HALF_AN_HOUR = 1
+private const val POS_AN_HOUR = 2
+private const val POS_FOUR_HOURS = 3
+private const val POS_TODAY = 4
+private const val POS_END_OF_WEEK = 5
+
+private const val ONE_SECOND_IN_MILLIS = 1000
+private const val ONE_MINUTE_IN_SECONDS = 60
+private const val THIRTY_MINUTES = 30
+private const val FOUR_HOURS = 4
+private const val LAST_HOUR_OF_DAY = 23
+private const val LAST_MINUTE_OF_HOUR = 59
+private const val LAST_SECOND_OF_MINUTE = 59
+
+@AutoInjector(NextcloudTalkApplication::class)
+class SetStatusDialogFragment :
+    DialogFragment(), PredefinedStatusClickListener {
+
+    private val logTag = ChooseAccountDialogFragment::class.java.simpleName
+
+
+    private lateinit var binding: DialogSetStatusBinding
+
+    private var currentUser: User? = null
+    private var currentStatus: Status? = null
+
+    // private lateinit var accountManager: UserAccountManager
+    // private lateinit var predefinedStatus: ArrayList<PredefinedStatus>
+    val predefinedStatusesList = ArrayList<PredefinedStatus>()
+
+    private lateinit var adapter: PredefinedStatusListAdapter
+    private var selectedPredefinedMessageId: String? = null
+    private var clearAt: Long? = -1
+    private lateinit var popup: EmojiPopup
+
+    // @Inject
+    // lateinit var arbitraryDataProvider: ArbitraryDataProvider
+    //
+    // @Inject
+    // lateinit var asyncRunner: AsyncRunner
+    //
+    // @Inject
+    // lateinit var clientFactory: ClientFactory
+
+    @Inject
+    lateinit var ncApi: NcApi
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            currentUser = it.getParcelable(ARG_CURRENT_USER_PARAM)
+            currentStatus = it.getParcelable(ARG_CURRENT_STATUS_PARAM)
+
+
+
+            val credentials = ApiUtils.getCredentials(currentUser?.username, currentUser?.token)
+            ncApi.getPredefinedStatuses(credentials, ApiUtils.getUrlForPredefinedStatuses(currentUser?.baseUrl))
```

Comment:
wondering why i get 
kotlin.UninitializedPropertyAccessException: lateinit property ncApi has not been initialized

anything wrong with the injection?
any ideas?  @tobiasKaminsky ?

Reply:
Not sure, but other activity have something like this
NextcloudTalkApplication.sharedApplication!!.componentApplication.inject(this)

Reply:
On dagger, which we use in Files, it is not possible to inject into Fragments, but only to Activities.
We then always pass this as an argument.

Reply:
ah great, thanks :+1:  i already suspected that it would have something to do with fragments..

Reply:
it was not about fragments, that works indeed. it was really just 
NextcloudTalkApplication.sharedApplication!!.componentApplication.inject(this)
that i forgot :D